src/native/eventpipe/ep-thread.h

```diff
@@ -45,6 +45,11 @@ struct _EventPipeThread_Internal {
 	// that pointer will be protected from deletion. See ep_disable () and
 	// ep_write () for more detail.
 	volatile uint32_t writing_event_in_progress;
+	// This is set to non-zero when the thread is unregistered from the global list of EventPipe threads.
+	// This should happen when a physical thread is ending.
+	// This is a convenience marker to prevent us from having to search the global list.
+	// defaults to false.
+	volatile uint32_t unregistered;
```

Comment:
if all we do to this value is set it to 0 or 1 can we use a smaller size type instead of uint32_t?

Reply:
Structure alignment means it would not make a difference in overall structure size if we used uint8_t, it would still be padded to a 4 byte or 8 byte boundary depending on 32/64 bit. It would only save us space if we had other bytes of data to go after.

If we were really concerned about the size of this struct we could steal a bit from some other data here. E.g. ref_count we could take the high bit, it seems unlikely we'd ever have more than 2^31 ref counts

 I don't think 4 more bytes is a big concern since it's tied 1-1 with threads, they have so much overhead already (managed Thread object, 1MB stack space by default, etc).